core/src/crds.rs

```diff
@@ -153,33 +168,30 @@ impl Crds {
                         self.votes.insert(entry_index);
                     }
                     CrdsData::EpochSlots(_, _) => {
-                        self.epoch_slots
-                            .insert((value.insert_timestamp, entry_index));
+                        self.epoch_slots.insert(value.ordinal, entry_index);
                     }
                     _ => (),
                 };
                 self.records
                     .entry(value.value.pubkey())
                     .or_default()
                     .insert(entry_index);
+                self.cursor.consume(value.ordinal);
```

Comment:
ah nice! this prevents us from losing values in calls like `get_votes(since)` if a new value with the same `timestamp == since` arrived since the last fetch.